docs/reverse-replication/ReverseReplicationUserGuide.md

```diff
@@ -243,12 +252,67 @@ Note: Additional optional parameters for the reader job are [here](https://githu
 
 Note: Additional optional parameters for the writer job are [here](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/gcs-to-sourcedb/README_GCS_to_Sourcedb.md#optional-parameters).
 
+### Ignorable errors
+
+Dataflow retries most of the errors. Following errors if shown up in the Dataflow UI can be ignored.
+
+#### Reader job
+
+1. File not found exception like below. These are thrown at the time Dataflow workers auto-scale and the work gets reassigned among the workers.
+
+```
+java.io.FileNotFoundException: Rewrite from <GCS bucket name>/.temp-beam/<file name> to <GCS file path> has failed
+```
+
+2. Spanner DEADLINE_EXCEEDED exception.
+
+3. GC thrashing exception like below
+
+```
+Shutting down JVM after 8 consecutive periods of measured GC thrashing.
+```
+
+4. GCS throttling can slow down writes for a while. Below exception results during that period and can be ignored.
+
+```
+Operation ongoing in bundle process_bundle-<bundle number> for PTransform{id=Write To GCS/Write rows to output writeDynamic/WriteFiles/WriteShardedBundlesToTempFiles/WriteShardsIntoTempFiles-ptransform-46, name=Write To GCS/Write rows to output writeDynamic/WriteFiles/WriteShardedBundlesToTempFiles/WriteShardsIntoTempFiles-ptransform-46, state=process} for at least 05m20s without outputting or completing
+```
+
+```
+java.io.IOException: Error executing batch GCS request
+```
+
+#### Writer job
+
+1. To preserve ordering, the writer job processes files in incrementing window intervals. If the reader job is lagging in creating files, the writer job waits for the expected file for a given window to be written to GCS. In such cases, below messages and logged and can be ignored.
+
+```
+Operation ongoing in step Write to source for at least 05m00s without outputting or completing in state process-timers in thread DataflowWorkUnits-11418 with id 1891593
+  at java.base@11.0.20/java.lang.Thread.sleep(Native Method)
+  at app//com.google.cloud.teleport.v2.templates.utils.GCSReader.checkAndReturnIfFileExists
+```
+
+2. Large amount of logging can result in below. This does not halt the Dataflow job from processing.
+
+```
+Throttling logger worker. It used up its 30s quota for logs in only 17.465s
+```
+
+#### Common to both jobs
+
+1. Ephemeral network glitches results in below ignorable error.
+
+```
+StatusRuntimeException: UNAVAILABLE: ping timeout
+```
+
 ## Reverse Replication Limitations
 
   The following sections list the known limitations that exist currently with the Reverse Replication flows:
 
   1. Currently only MySQL source database is supported.
-  2. Certain transformations are not supported, below section lists those:
+  2. If forward migration and reverse replication are running in parallel, there is no mechanism to prevent the forward migration of data that was written to source via the reverse replicaiton flow. The impact of this is unnecessary processing of redundant data.
```

Comment:
typo: replication*

Reply:
We do filter the transactions done via the forward migration flow in the filtering step. That would prevent the fwd migration data right.

What we do not handle is post cutover, continuing running the forward pipeline can cause processing of redundant data and temporary inconsistencies on spanner. We should also recommend stopping the forward pipeline post cutover asap.

Reply:
done. Also added to the best practices section.